pkg/engine2/operational_eval/vertex_path_expand.go

```diff
@@ -584,16 +579,20 @@ func getDepsForPropertyRef(
 	res construct.ResourceId,
 	propertyRef string,
 ) set.Set[Key] {
+	if propertyRef == "" {
+		return nil
+	}
 	keys := make(set.Set[Key])
 	cfgCtx := solution_context.DynamicCtx(sol)
 	currResources := []construct.ResourceId{res}
 	parts := strings.Split(propertyRef, "#")
 	for _, part := range parts {
 		var nextResources []construct.ResourceId
 		for _, currResource := range currResources {
+			keys.Add(Key{Ref: construct.PropertyRef{Resource: currResource, Property: part}})
```

Comment:
This is the other part of the fix, all elements of the property path need to be depended on.